RecoEgamma/EgammaHLTProducers/plugins/EgammaHLTGsfTrackVarProducer.cc

```diff
@@ -254,6 +274,21 @@ void EgammaHLTGsfTrackVarProducer::produce(edm::StreamID, edm::Event& iEvent, co
   iEvent.emplace(validHitsMapPutToken_, validHitsMap);
   iEvent.emplace(nLayerITMapPutToken_, nLayerITMap);
   iEvent.emplace(chi2MapPutToken_, chi2Map);
+  iEvent.emplace(fbremMapPutToken_, fbremMap);
+}
+
+void EgammaHLTGsfTrackVarProducer::GsfTrackExtrapolations::operator()(
+    const reco::GsfTrack& trk, const reco::SuperCluster& sc, const MultiTrajectoryStateTransform& mtsTransform) {
+  innTSOS = mtsTransform.innerStateOnSurface(trk);
+  outTSOS = mtsTransform.outerStateOnSurface(trk);
+  sclTSOS = mtsTransform.extrapolatedState(innTSOS, GlobalPoint(sc.x(), sc.y(), sc.z()));
+
+  multiTrajectoryStateMode::momentumFromModeCartesian(innTSOS, innMom);
+  multiTrajectoryStateMode::positionFromModeCartesian(innTSOS, innPos);
+  multiTrajectoryStateMode::momentumFromModeCartesian(sclTSOS, sclMom);
+  multiTrajectoryStateMode::positionFromModeCartesian(sclTSOS, sclPos);
+  multiTrajectoryStateMode::momentumFromModeCartesian(outTSOS, outMom);
+  multiTrajectoryStateMode::positionFromModeCartesian(outTSOS, outPos);
```

Comment:
If I checked correctly, `innPos`, `sclMom` and `outPos` are not really used in the plugin. If so, should one avoid computing them, or is it very inexpensive to compute them anyway ?

Reply:
yes I should check this and run a timing study.  It was mostly a copy and paste from the GsfElectron class, I was debating storing them earlier  and forgot to delete them, thanks